lib/sycamore/sycamore/tests/unit/connectors/opensearch/test_opensearch.py

```diff
@@ -461,3 +466,56 @@ def test_get_knn_query(self):
     def test_get_knn_query_validation(self):
         with pytest.raises(ValueError, match="Only one of `k` or `min_score` should be populated"):
             get_knn_query(Mock(spec=Embedder), query_phrase="test", k=10, min_score=0.5)
+
+
+class TestOpenSearchReader:
+
+    def test_add_filter_to_query(self):
```

Comment:
I'd make a stronger test:
```
assert query == { "query": { "bool": { "must": [{"match_all": {}}], "filter": [{"terms": {"properties.colors": ["red", "blue"] }
```

It's shorter, and it's verifying exactly what you want to get so can catch more bugs.